bgs/compactor.go

```diff
@@ -217,30 +252,29 @@ func (c *Compactor) Start(bgs *BGS) {
 func (c *Compactor) Shutdown() {
 	log.Info("stopping compactor")
 	close(c.exit)
-	<-c.exited
+	c.wg.Wait()
 	log.Info("compactor stopped")
 }
 
-func (c *Compactor) doWork(bgs *BGS) {
+func (c *Compactor) doWork(bgs *BGS, wg *sync.WaitGroup, strategy NextStrategy) {
```

Comment:
no need to pass the waitgroup, just use the struct member

Reply:
I think the change here is to drop the `Compactor.exited` member in favor of a sync.WorkGroup which more naturally fits N workers instead of 1

Reply:
The waitgroup is on the `*Compactor` object already though isn't it? It doesn't need to be passed into this function as well as a param

Reply:
derp. yes.